src/pykiso/lib/connectors/cc_serial.py

```diff
@@ -119,7 +125,60 @@ def __init__(
         )
 
         self.current_write_timeout = write_timeout
-        self.serial.port = port
+        self.serial.port = self._get_port(port=port, vid=vid, pid=pid, timeout=0.1)
+
+    def _find_device(vid: int, pid: int, timeout: float = 0.1) -> str:
+        """Return the device which matches pid and vid.
+
+        :param vid: vendor id
+        :param pid: product id
+        :param timeout: timeout in seconds to wait for at least one device
+            to be found.
+        :return: com port for the found device. I.e. "COM4" or "/dev/tty1"
+        """
+        start_time = time.perf_counter()
+
+        while time.perf_counter() - start_time < timeout:
+            attached_com_devices = serial.tools.list_ports.comports()
+            found_devices = [
+                port
+                for port in attached_com_devices
+                if port.pid == pid and port.vid == vid
+            ]
+            if found_devices:
+                break
+            time.sleep(0.1)
```

Comment:
Currently those timeouts makes no sense..
You scan once and wait 100 ms. A second scan wont be triggered so that we waste always 100 ms .

I think we dont need a wait for  HW to be plugged in general.. 

HW should be always connected attached when we start a kiso test

Reply:
done